src/shared/xml.interceptor.ts

```diff
@@ -0,0 +1,15 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common';
+import * as jsontoxml from 'jsontoxml';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class XMLInterceptor<T> implements NestInterceptor<T, string> {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<string> {
+    const ctx = context.switchToHttp();
+    const request = ctx.getRequest<Request>();
+    const requestedXML = request.headers['content-type'] === 'text/xml';
```

Comment:
How about checking for `application/xml` instead of `text/xml`, or both?

Reply:
Fixed!